---
title: "Illustrative examples"
author: "Xiaotong Sun, Gao Wang, Shengtong Han"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{minimal example}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(collapse = TRUE,comment = "#",fig.width = 4.5,
                      fig.height = 3,fig.align = "center",
                      fig.cap = "&nbsp;",dpi = 120)
```

## Installation

```{r install_package, eval=F}
devtools::install_github('xinhe-lab/mirage')
```

Load the package

```{r load_package}
library(mirage)
```


## Preparing input data

### Summary statistics for each rare variant. 

MIRAGE requires summary statistics as input, including the ID for each rare variant. For each variant, it needs the allele counts in both cases and controls, the gene to which it belongs, and the annotation group index it is associated with. The annotation groups should be distinct categories, such as loss-of-function (LoF) variants and missense variants predicted to have deleterious effects.

We include an example input for MIRAGE in our package, which users can reference to format their own data accordingly.


```{r}
data(mirage_toy)
head(mirage_toy)
```

Each row in the input represents a variant with its summary statistics:


- `ID`: variant ID
- `No.case`: Number of times the variant appears in cases.
- `No.contr`: Number of times the variant appears in controls.
- `group.index`: Indicates the variant group it belongs to.
- `Gene`: Specifies the associated gene, if any.

**Please note that, the analysis is limited to rare variants. So the MAF (minor allele frequency) of all variants used in the analysis should be less than 0.01**

In the example below, since the sample size is (4315 + 4315, case + control), users can filter the variants using 

`mirage_toy[mirage_toy$No.case/(4315 + 4315) < 0.01|mirage_toy$No.contr/(4315 + 4315) < 0.01,]`

Variant groups are user-defined, usually based on annotations. For example, a group "rare LoF" could be defined for loss-of-function variants (including stop loss, stop gain, frameshift indels, and splice site substitutions) with a minor allele frequency of less than 1%.



## Additional arguments

MIRAGE requires the following additional arguments:

- `n1`: Sample size in cases (required).
- `n2`: Sample size in controls (required).
- `gamma`: A list of category-specific hyper prior shape parameters in the Beta distribution for effect size, or a numeric value if all categories share the same effect size. Default value is 3.
- `sigma`: A list of category-specific hyper prior scale parameters in the Beta distribution for effect size, or a numeric value if all categories share the same effect size. Default value is 2.
- `eta.init`: Initial value for the prior on the proportion of risk variants in a variant set. Default value is 0.1.
- `delta.init`: Initial value for the prior on the proportion of risk genes. Must be a positive number between 0 and 1. Default value is 0.1.
- `estimate.delta`: Specifies whether `delta` should be estimated (`TRUE`) or fixed at `delta.init` (`FALSE`). Default is `TRUE`.
- `estimate.eta` Specifies where `eta` should be estimated (`TRUE`) or fixed at  `fixed.eta`. Default is `TRUE`. 
- `fixed.eta` Must be provided when `estimate.eta` is false and   BF per gene will be reported. **Its dimension must match the number of variant groups**. 
- `max.iter`: Maximum number of iterations, enforcing the EM algorithm to stop. Default is 10,000.
- `tol`: Threshold of parameter estimate difference to determine the convergence of the EM algorithm. Default value is $10^{-5}$.
- `verbose`: When `TRUE`, enables detailed output during computation. Default is `TRUE`.

Only `n1` and `n2` are required to run MIRAGE; other parameters have default values.

## Running MIRAGE on toy data

```{r, cache=T}
res=mirage(mirage_toy,n1=4315,n2=4315)
```

The output of MIRAGE is a list containing the following values:

- `BF.PP.gene`: Bayes factor and posterior probability of genes.

```{r }
head(res$BF.PP.gene)
```

- `BF.all`: A list of Bayes factors for all variants within a gene.

```{r }
head(res$BF.all[1],n = 5) # variant information in the first gene 
```

- `delta.est`: Estimate for the proportion of risk genes and p value.

```{r }
res$delta.est
```


- `eta.est`: Estimate for the proportion of risk variants in each variant group and p values.

```{r }
res$eta.est
```


### Calculating Bayes factor for each variant group within a gene 

Users may use the script below to compute Bayes factor for each variant group within a gene. We here show an example for computing for the first gene in our example data "LPA". 


```{r, cache=T}
########## after run MIRAGE,  given a gene, calculate the BF of every variant group.
gene_name <- "LPA"

# Number of variant groups for this gene
num_category <- nlevels(factor(mirage_toy$group.index[mirage_toy$Gene == gene_name]))

# Extract variant-level results for this gene
variant_full_infor <- res$BF.all[[which(res$BF.PP.gene$Gene == gene_name)]]

# Compute group-level BF
variant_group_BF <- sapply(seq_len(num_category), function(j) {
  idx <- which(variant_full_infor$original.index == j)
  if (length(idx) == 0) {
    return(NA_real_)
  } else {
    return(prod(1 - res$eta.est[j] + res$eta.est[j] * variant_full_infor$var.BF[idx]))
  }
})

variant_group_BF

```

### Calculating Bayes factor with fixed eta


```{r, cache=T}
res_fixed_eta=mirage(mirage_toy,n1=4315,n2=4315, estimate.eta = F, fixed.eta = c(0.60, 0.31, 0.16, 0.4, 0.2))
```

The output of MIRAGE is a list containing the following values:

- `BF.gene`: a matrix of gene with BF's 

```{r }
head(res_fixed_eta$BF.gene)
```


- `BF.all`: A list of Bayes factors for all variants within a gene.

```{r }
head(res_fixed_eta$BF.all[1], n=5) # variant information in the first gene 
```
















## Session information

Here are some details about the computing environment, including the
versions of R, and the R packages, used to generate these results.

```{r}
sessionInfo()
```
