<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Xiaotong Sun, Gao Wang, Shengtong Han" />

<meta name="date" content="2025-09-05" />

<title>Illustrative examples</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Illustrative examples</h1>
<h4 class="author">Xiaotong Sun, Gao Wang, Shengtong Han</h4>
<h4 class="date">2025-09-05</h4>



<div id="installation" class="section level2">
<h2>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&#39;xinhe-lab/mirage&#39;</span>)</span></code></pre></div>
<p>Load the package</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(mirage)</span></code></pre></div>
</div>
<div id="preparing-input-data" class="section level2">
<h2>Preparing input data</h2>
<div id="summary-statistics-for-each-rare-variant." class="section level3">
<h3>Summary statistics for each rare variant.</h3>
<p>MIRAGE requires summary statistics as input, including the ID for
each rare variant. For each variant, it needs the allele counts in both
cases and controls, the gene to which it belongs, and the annotation
group index it is associated with. The annotation groups should be
distinct categories, such as loss-of-function (LoF) variants and
missense variants predicted to have deleterious effects.</p>
<p>We include an example input for MIRAGE in our package, which users
can reference to format their own data accordingly.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">data</span>(mirage_toy)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">head</span>(mirage_toy)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#                                 ID   Gene No.case No.contr group.index</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co"># 15129    6:161006077-161006077_C_T    LPA     298      272           1</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co"># 169        1:11205024-11205024_C_T   MTOR       1        0           2</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co"># 184        1:11272449-11272449_G_A   MTOR       0        1           2</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co"># 194   1:11290989-11290994_CTGACT_C   MTOR       1        0           2</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co"># 239        1:11594465-11594465_C_T PTCHD2       0        1           2</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co"># 356     1:19441304-19441307_CCTT_C   UBR4       2        0           2</span></span></code></pre></div>
<p>Each row in the input represents a variant with its summary
statistics:</p>
<ul>
<li><code>ID</code>: variant ID</li>
<li><code>No.case</code>: Number of times the variant appears in
cases.</li>
<li><code>No.contr</code>: Number of times the variant appears in
controls.</li>
<li><code>group.index</code>: Indicates the variant group it belongs
to.</li>
<li><code>Gene</code>: Specifies the associated gene, if any.</li>
</ul>
<p><strong>Please note that, the analysis is limited to rare variants.
So the MAF (minor allele frequency) of all variants used in the analysis
should be less than 0.01</strong></p>
<p>In the example below, since the sample size is (4315 + 4315, case +
control), users can filter the variants using</p>
<p><code>mirage_toy[mirage_toy$No.case/(4315 + 4315) &lt; 0.01|mirage_toy$No.contr/(4315 + 4315) &lt; 0.01,]</code></p>
<p>Variant groups are user-defined, usually based on annotations. For
example, a group “rare LoF” could be defined for loss-of-function
variants (including stop loss, stop gain, frameshift indels, and splice
site substitutions) with a minor allele frequency of less than 1%.</p>
</div>
</div>
<div id="additional-arguments" class="section level2">
<h2>Additional arguments</h2>
<p>MIRAGE requires the following additional arguments:</p>
<ul>
<li><code>n1</code>: Sample size in cases (required).</li>
<li><code>n2</code>: Sample size in controls (required).</li>
<li><code>gamma</code>: A list of category-specific hyper prior shape
parameters in the Beta distribution for effect size, or a numeric value
if all categories share the same effect size. Default value is 3.</li>
<li><code>sigma</code>: A list of category-specific hyper prior scale
parameters in the Beta distribution for effect size, or a numeric value
if all categories share the same effect size. Default value is 2.</li>
<li><code>eta.init</code>: Initial value for the prior on the proportion
of risk variants in a variant set. Default value is 0.1.</li>
<li><code>delta.init</code>: Initial value for the prior on the
proportion of risk genes. Must be a positive number between 0 and 1.
Default value is 0.1.</li>
<li><code>estimate.delta</code>: Specifies whether <code>delta</code>
should be estimated (<code>TRUE</code>) or fixed at
<code>delta.init</code> (<code>FALSE</code>). Default is
<code>TRUE</code>.</li>
<li><code>estimate.eta</code> Specifies where <code>eta</code> should be
estimated (<code>TRUE</code>) or fixed at <code>fixed.eta</code>.
Default is <code>TRUE</code>.</li>
<li><code>fixed.eta</code> Must be provided when
<code>estimate.eta</code> is false and BF per gene will be reported.
<strong>Its dimension must match the number of variant
groups</strong>.</li>
<li><code>max.iter</code>: Maximum number of iterations, enforcing the
EM algorithm to stop. Default is 10,000.</li>
<li><code>tol</code>: Threshold of parameter estimate difference to
determine the convergence of the EM algorithm. Default value is <span class="math inline">\(10^{-5}\)</span>.</li>
<li><code>verbose</code>: When <code>TRUE</code>, enables detailed
output during computation. Default is <code>TRUE</code>.</li>
</ul>
<p>Only <code>n1</code> and <code>n2</code> are required to run MIRAGE;
other parameters have default values.</p>
</div>
<div id="running-mirage-on-toy-data" class="section level2">
<h2>Running MIRAGE on toy data</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>res<span class="ot">=</span><span class="fu">mirage</span>(mirage_toy,<span class="at">n1=</span><span class="dv">4315</span>,<span class="at">n2=</span><span class="dv">4315</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co"># Computing gene level LRT statistics and p-values ...</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co"># Computing LRT statistics and p-values by categories ...</span></span></code></pre></div>
<p>The output of MIRAGE is a list containing the following values:</p>
<ul>
<li><code>BF.PP.gene</code>: Bayes factor and posterior probability of
genes.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">head</span>(res<span class="sc">$</span>BF.PP.gene)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co">#     Gene        BF post.prob</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co"># 1    LPA 0.9164196 0.9190893</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co"># 2   MTOR 1.0405812 0.9280489</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co"># 3 PTCHD2 0.8327124 0.9116743</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co"># 4   UBR4 1.2899155 0.9411381</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co"># 5  CSMD2 0.8010560 0.9085032</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co"># 6   NCDN 1.0006285 0.9253904</span></span></code></pre></div>
<ul>
<li><code>BF.all</code>: A list of Bayes factors for all variants within
a gene.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">head</span>(res<span class="sc">$</span>BF.all[<span class="dv">1</span>],<span class="at">n =</span> <span class="dv">5</span>) <span class="co"># variant information in the first gene </span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co"># [[1]]</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#                                ID Gene No.case No.contr category</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co"># 15129   6:161006077-161006077_C_T  LPA     298      272        1</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co"># 15078 6:160966536-160966536_A_ACC  LPA       2        2        2</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co"># 15079 6:160966537-160966539_TTG_T  LPA       2        2        2</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co"># 15082  6:160968821-160968822_AT_A  LPA       1        0        2</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co"># 15102   6:160969584-160969584_G_C  LPA       1        4        2</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co"># 15107  6:160977184-160977185_AC_A  LPA       0        1        2</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co"># 15114   6:160978560-160978560_G_A  LPA       1        0        2</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co"># 15116   6:160998167-160998167_C_T  LPA       1        5        2</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co"># 15142   6:161007480-161007480_C_T  LPA       1        0        2</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="co"># 15153   6:161010638-161010638_C_T  LPA       1        0        2</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="co"># 15163   6:161012014-161012014_G_T  LPA       0        1        2</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="co"># 15171   6:161015102-161015102_C_A  LPA       1        0        2</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co"># 15177  6:161016494-161016494_C_CA  LPA       1        0        2</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="co"># 15180   6:161020531-161020531_C_T  LPA       0        1        2</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="co"># 15210   6:161071369-161071369_C_T  LPA       1        0        2</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co"># 15061   6:160952816-160952816_T_C  LPA     118      103        3</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="co"># 15068   6:160961137-160961137_T_C  LPA     257      248        3</span></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="co"># 15073   6:160963771-160963771_C_A  LPA      50       44        4</span></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a><span class="co"># 15074   6:160963774-160963774_C_G  LPA      21       32        4</span></span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a><span class="co"># 15075   6:160963774-160963774_C_T  LPA       0        2        4</span></span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a><span class="co"># 15081   6:160966559-160966559_G_A  LPA      17       28        4</span></span>
<span id="cb6-25"><a href="#cb6-25" tabindex="-1"></a><span class="co"># 15135   6:161006105-161006105_C_T  LPA      88      100        4</span></span>
<span id="cb6-26"><a href="#cb6-26" tabindex="-1"></a><span class="co"># 15148   6:161007619-161007619_G_T  LPA      26       20        4</span></span>
<span id="cb6-27"><a href="#cb6-27" tabindex="-1"></a><span class="co"># 15195   6:161022107-161022107_C_T  LPA      79       90        4</span></span>
<span id="cb6-28"><a href="#cb6-28" tabindex="-1"></a><span class="co">#       category_factor       var.BF original.index</span></span>
<span id="cb6-29"><a href="#cb6-29" tabindex="-1"></a><span class="co"># 15129               1 1.634504e-07              1</span></span>
<span id="cb6-30"><a href="#cb6-30" tabindex="-1"></a><span class="co"># 15078               2 6.059431e-01              2</span></span>
<span id="cb6-31"><a href="#cb6-31" tabindex="-1"></a><span class="co"># 15079               2 6.059431e-01              2</span></span>
<span id="cb6-32"><a href="#cb6-32" tabindex="-1"></a><span class="co"># 15082               2 1.452084e+00              2</span></span>
<span id="cb6-33"><a href="#cb6-33" tabindex="-1"></a><span class="co"># 15102               2 1.896752e-01              2</span></span>
<span id="cb6-34"><a href="#cb6-34" tabindex="-1"></a><span class="co"># 15107               2 5.479161e-01              2</span></span>
<span id="cb6-35"><a href="#cb6-35" tabindex="-1"></a><span class="co"># 15114               2 1.452084e+00              2</span></span>
<span id="cb6-36"><a href="#cb6-36" tabindex="-1"></a><span class="co"># 15116               2 1.405388e-01              2</span></span>
<span id="cb6-37"><a href="#cb6-37" tabindex="-1"></a><span class="co"># 15142               2 1.452084e+00              2</span></span>
<span id="cb6-38"><a href="#cb6-38" tabindex="-1"></a><span class="co"># 15153               2 1.452084e+00              2</span></span>
<span id="cb6-39"><a href="#cb6-39" tabindex="-1"></a><span class="co"># 15163               2 5.479161e-01              2</span></span>
<span id="cb6-40"><a href="#cb6-40" tabindex="-1"></a><span class="co"># 15171               2 1.452084e+00              2</span></span>
<span id="cb6-41"><a href="#cb6-41" tabindex="-1"></a><span class="co"># 15177               2 1.452084e+00              2</span></span>
<span id="cb6-42"><a href="#cb6-42" tabindex="-1"></a><span class="co"># 15180               2 5.479161e-01              2</span></span>
<span id="cb6-43"><a href="#cb6-43" tabindex="-1"></a><span class="co"># 15210               2 1.452084e+00              2</span></span>
<span id="cb6-44"><a href="#cb6-44" tabindex="-1"></a><span class="co"># 15061               3 7.596685e-02              3</span></span>
<span id="cb6-45"><a href="#cb6-45" tabindex="-1"></a><span class="co"># 15068               3 3.931896e-08              3</span></span>
<span id="cb6-46"><a href="#cb6-46" tabindex="-1"></a><span class="co"># 15073               4 9.300675e-02              4</span></span>
<span id="cb6-47"><a href="#cb6-47" tabindex="-1"></a><span class="co"># 15074               4 5.040937e-02              4</span></span>
<span id="cb6-48"><a href="#cb6-48" tabindex="-1"></a><span class="co"># 15075               4 3.291740e-01              4</span></span>
<span id="cb6-49"><a href="#cb6-49" tabindex="-1"></a><span class="co"># 15081               4 5.484098e-02              4</span></span>
<span id="cb6-50"><a href="#cb6-50" tabindex="-1"></a><span class="co"># 15135               4 2.698210e-02              4</span></span>
<span id="cb6-51"><a href="#cb6-51" tabindex="-1"></a><span class="co"># 15148               4 2.857868e-01              4</span></span>
<span id="cb6-52"><a href="#cb6-52" tabindex="-1"></a><span class="co"># 15195               4 2.795875e-02              4</span></span></code></pre></div>
<ul>
<li><code>delta.est</code>: Estimate for the proportion of risk
genes.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>res<span class="sc">$</span>delta.est</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="co"># [1] 0.925347</span></span></code></pre></div>
<ul>
<li><code>delta.pvalue</code>: Significance test for delta = 0.</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>res<span class="sc">$</span>delta.pvalue</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co"># [1] 0.9347734</span></span></code></pre></div>
<ul>
<li><code>eta.est</code>: Estimate for the proportion of risk variants
in a variant group.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>res<span class="sc">$</span>eta.est</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co">#             1             2             3             4             5 </span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="co">#  0.000000e+00  1.030961e-01 4.940656e-324 1.976263e-323  7.871498e-02</span></span></code></pre></div>
<ul>
<li><code>eta.pvalue</code>: Significance test for eta = 0.</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>res<span class="sc">$</span>eta.pvalue</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="co"># [1] 1.0000000 0.4206792 1.0000000 1.0000000 0.2782136</span></span></code></pre></div>
<div id="calculating-bayes-factor-for-each-variant-group-within-a-gene" class="section level3">
<h3>Calculating Bayes factor for each variant group within a gene</h3>
<p>Users may use the script below to compute Bayes factor for each
variant group within a gene. We here show an example for computing for
the first gene in our example data “LPA”.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="do">########## after run MIRAGE,  given a gene, calculate the BF of every variant group.</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>gene_name <span class="ot">&lt;-</span> <span class="st">&quot;LPA&quot;</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co"># Number of variant groups for this gene</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>num_category <span class="ot">&lt;-</span> <span class="fu">nlevels</span>(<span class="fu">factor</span>(mirage_toy<span class="sc">$</span>group.index[mirage_toy<span class="sc">$</span>Gene <span class="sc">==</span> gene_name]))</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co"># Extract variant-level results for this gene</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>variant_full_infor <span class="ot">&lt;-</span> res<span class="sc">$</span>BF.all[[<span class="fu">which</span>(res<span class="sc">$</span>BF.PP.gene<span class="sc">$</span>Gene <span class="sc">==</span> gene_name)]]</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co"># Compute group-level BF</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a>variant_group_BF <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq_len</span>(num_category), <span class="cf">function</span>(j) {</span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a>  idx <span class="ot">&lt;-</span> <span class="fu">which</span>(variant_full_infor<span class="sc">$</span>original.index <span class="sc">==</span> j)</span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(idx) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a>    <span class="fu">return</span>(<span class="cn">NA_real_</span>)</span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">prod</span>(<span class="dv">1</span> <span class="sc">-</span> res<span class="sc">$</span>eta.est[j] <span class="sc">+</span> res<span class="sc">$</span>eta.est[j] <span class="sc">*</span> variant_full_infor<span class="sc">$</span>var.BF[idx]))</span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a>  }</span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a>})</span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a></span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a>variant_group_BF</span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a><span class="co"># [1] 1.0000000 0.9164266 1.0000000 1.0000000</span></span></code></pre></div>
</div>
<div id="calculating-bayes-factor-with-fixed-eta" class="section level3">
<h3>Calculating Bayes factor with fixed eta</h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>res_fixed_eta<span class="ot">=</span><span class="fu">mirage</span>(mirage_toy,<span class="at">n1=</span><span class="dv">4315</span>,<span class="at">n2=</span><span class="dv">4315</span>, <span class="at">estimate.eta =</span> F, <span class="at">fixed.eta =</span> <span class="fu">c</span>(<span class="fl">0.60</span>, <span class="fl">0.31</span>, <span class="fl">0.16</span>, <span class="fl">0.4</span>, <span class="fl">0.2</span>))</span></code></pre></div>
<p>The output of MIRAGE is a list containing the following values:</p>
<ul>
<li><code>BF.gene</code>: a matrix of gene with BF’s</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">head</span>(res_fixed_eta<span class="sc">$</span>BF.gene)</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a><span class="co">#     Gene          BF</span></span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="co"># 1    LPA 0.009253237</span></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="co"># 2   MTOR 0.984853873</span></span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a><span class="co"># 3 PTCHD2 0.585846104</span></span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a><span class="co"># 4   UBR4 1.335284217</span></span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a><span class="co"># 5  CSMD2 0.077841990</span></span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a><span class="co"># 6   NCDN 1.023760609</span></span></code></pre></div>
<ul>
<li><code>BF.all</code>: A list of Bayes factors for all variants within
a gene.</li>
</ul>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">head</span>(res_fixed_eta<span class="sc">$</span>BF.all[<span class="dv">1</span>], <span class="at">n=</span><span class="dv">5</span>) <span class="co"># variant information in the first gene </span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="co"># [[1]]</span></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="co">#                                ID Gene No.case No.contr category</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="co"># 15129   6:161006077-161006077_C_T  LPA     298      272        1</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a><span class="co"># 15078 6:160966536-160966536_A_ACC  LPA       2        2        2</span></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a><span class="co"># 15079 6:160966537-160966539_TTG_T  LPA       2        2        2</span></span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a><span class="co"># 15082  6:160968821-160968822_AT_A  LPA       1        0        2</span></span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a><span class="co"># 15102   6:160969584-160969584_G_C  LPA       1        4        2</span></span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a><span class="co"># 15107  6:160977184-160977185_AC_A  LPA       0        1        2</span></span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a><span class="co"># 15114   6:160978560-160978560_G_A  LPA       1        0        2</span></span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a><span class="co"># 15116   6:160998167-160998167_C_T  LPA       1        5        2</span></span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a><span class="co"># 15142   6:161007480-161007480_C_T  LPA       1        0        2</span></span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a><span class="co"># 15153   6:161010638-161010638_C_T  LPA       1        0        2</span></span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a><span class="co"># 15163   6:161012014-161012014_G_T  LPA       0        1        2</span></span>
<span id="cb14-15"><a href="#cb14-15" tabindex="-1"></a><span class="co"># 15171   6:161015102-161015102_C_A  LPA       1        0        2</span></span>
<span id="cb14-16"><a href="#cb14-16" tabindex="-1"></a><span class="co"># 15177  6:161016494-161016494_C_CA  LPA       1        0        2</span></span>
<span id="cb14-17"><a href="#cb14-17" tabindex="-1"></a><span class="co"># 15180   6:161020531-161020531_C_T  LPA       0        1        2</span></span>
<span id="cb14-18"><a href="#cb14-18" tabindex="-1"></a><span class="co"># 15210   6:161071369-161071369_C_T  LPA       1        0        2</span></span>
<span id="cb14-19"><a href="#cb14-19" tabindex="-1"></a><span class="co"># 15061   6:160952816-160952816_T_C  LPA     118      103        3</span></span>
<span id="cb14-20"><a href="#cb14-20" tabindex="-1"></a><span class="co"># 15068   6:160961137-160961137_T_C  LPA     257      248        3</span></span>
<span id="cb14-21"><a href="#cb14-21" tabindex="-1"></a><span class="co"># 15073   6:160963771-160963771_C_A  LPA      50       44        4</span></span>
<span id="cb14-22"><a href="#cb14-22" tabindex="-1"></a><span class="co"># 15074   6:160963774-160963774_C_G  LPA      21       32        4</span></span>
<span id="cb14-23"><a href="#cb14-23" tabindex="-1"></a><span class="co"># 15075   6:160963774-160963774_C_T  LPA       0        2        4</span></span>
<span id="cb14-24"><a href="#cb14-24" tabindex="-1"></a><span class="co"># 15081   6:160966559-160966559_G_A  LPA      17       28        4</span></span>
<span id="cb14-25"><a href="#cb14-25" tabindex="-1"></a><span class="co"># 15135   6:161006105-161006105_C_T  LPA      88      100        4</span></span>
<span id="cb14-26"><a href="#cb14-26" tabindex="-1"></a><span class="co"># 15148   6:161007619-161007619_G_T  LPA      26       20        4</span></span>
<span id="cb14-27"><a href="#cb14-27" tabindex="-1"></a><span class="co"># 15195   6:161022107-161022107_C_T  LPA      79       90        4</span></span>
<span id="cb14-28"><a href="#cb14-28" tabindex="-1"></a><span class="co">#       category_factor       var.BF original.index</span></span>
<span id="cb14-29"><a href="#cb14-29" tabindex="-1"></a><span class="co"># 15129               1 1.634504e-07              1</span></span>
<span id="cb14-30"><a href="#cb14-30" tabindex="-1"></a><span class="co"># 15078               2 6.059431e-01              2</span></span>
<span id="cb14-31"><a href="#cb14-31" tabindex="-1"></a><span class="co"># 15079               2 6.059431e-01              2</span></span>
<span id="cb14-32"><a href="#cb14-32" tabindex="-1"></a><span class="co"># 15082               2 1.452084e+00              2</span></span>
<span id="cb14-33"><a href="#cb14-33" tabindex="-1"></a><span class="co"># 15102               2 1.896752e-01              2</span></span>
<span id="cb14-34"><a href="#cb14-34" tabindex="-1"></a><span class="co"># 15107               2 5.479161e-01              2</span></span>
<span id="cb14-35"><a href="#cb14-35" tabindex="-1"></a><span class="co"># 15114               2 1.452084e+00              2</span></span>
<span id="cb14-36"><a href="#cb14-36" tabindex="-1"></a><span class="co"># 15116               2 1.405388e-01              2</span></span>
<span id="cb14-37"><a href="#cb14-37" tabindex="-1"></a><span class="co"># 15142               2 1.452084e+00              2</span></span>
<span id="cb14-38"><a href="#cb14-38" tabindex="-1"></a><span class="co"># 15153               2 1.452084e+00              2</span></span>
<span id="cb14-39"><a href="#cb14-39" tabindex="-1"></a><span class="co"># 15163               2 5.479161e-01              2</span></span>
<span id="cb14-40"><a href="#cb14-40" tabindex="-1"></a><span class="co"># 15171               2 1.452084e+00              2</span></span>
<span id="cb14-41"><a href="#cb14-41" tabindex="-1"></a><span class="co"># 15177               2 1.452084e+00              2</span></span>
<span id="cb14-42"><a href="#cb14-42" tabindex="-1"></a><span class="co"># 15180               2 5.479161e-01              2</span></span>
<span id="cb14-43"><a href="#cb14-43" tabindex="-1"></a><span class="co"># 15210               2 1.452084e+00              2</span></span>
<span id="cb14-44"><a href="#cb14-44" tabindex="-1"></a><span class="co"># 15061               3 7.596685e-02              3</span></span>
<span id="cb14-45"><a href="#cb14-45" tabindex="-1"></a><span class="co"># 15068               3 3.931896e-08              3</span></span>
<span id="cb14-46"><a href="#cb14-46" tabindex="-1"></a><span class="co"># 15073               4 9.300675e-02              4</span></span>
<span id="cb14-47"><a href="#cb14-47" tabindex="-1"></a><span class="co"># 15074               4 5.040937e-02              4</span></span>
<span id="cb14-48"><a href="#cb14-48" tabindex="-1"></a><span class="co"># 15075               4 3.291740e-01              4</span></span>
<span id="cb14-49"><a href="#cb14-49" tabindex="-1"></a><span class="co"># 15081               4 5.484098e-02              4</span></span>
<span id="cb14-50"><a href="#cb14-50" tabindex="-1"></a><span class="co"># 15135               4 2.698210e-02              4</span></span>
<span id="cb14-51"><a href="#cb14-51" tabindex="-1"></a><span class="co"># 15148               4 2.857868e-01              4</span></span>
<span id="cb14-52"><a href="#cb14-52" tabindex="-1"></a><span class="co"># 15195               4 2.795875e-02              4</span></span></code></pre></div>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<p>Here are some details about the computing environment, including the
versions of R, and the R packages, used to generate these results.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="co"># R version 4.3.2 (2023-10-31 ucrt)</span></span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a><span class="co"># Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="co"># Running under: Windows 11 x64 (build 26100)</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a><span class="co"># Matrix products: default</span></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a><span class="co"># locale:</span></span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a><span class="co"># [1] LC_COLLATE=English_United States.utf8 </span></span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a><span class="co"># [2] LC_CTYPE=English_United States.utf8   </span></span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a><span class="co"># [3] LC_MONETARY=English_United States.utf8</span></span>
<span id="cb15-13"><a href="#cb15-13" tabindex="-1"></a><span class="co"># [4] LC_NUMERIC=C                          </span></span>
<span id="cb15-14"><a href="#cb15-14" tabindex="-1"></a><span class="co"># [5] LC_TIME=English_United States.utf8    </span></span>
<span id="cb15-15"><a href="#cb15-15" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb15-16"><a href="#cb15-16" tabindex="-1"></a><span class="co"># time zone: America/Chicago</span></span>
<span id="cb15-17"><a href="#cb15-17" tabindex="-1"></a><span class="co"># tzcode source: internal</span></span>
<span id="cb15-18"><a href="#cb15-18" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb15-19"><a href="#cb15-19" tabindex="-1"></a><span class="co"># attached base packages:</span></span>
<span id="cb15-20"><a href="#cb15-20" tabindex="-1"></a><span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb15-21"><a href="#cb15-21" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb15-22"><a href="#cb15-22" tabindex="-1"></a><span class="co"># other attached packages:</span></span>
<span id="cb15-23"><a href="#cb15-23" tabindex="-1"></a><span class="co"># [1] mirage_0.1.0.0</span></span>
<span id="cb15-24"><a href="#cb15-24" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb15-25"><a href="#cb15-25" tabindex="-1"></a><span class="co"># loaded via a namespace (and not attached):</span></span>
<span id="cb15-26"><a href="#cb15-26" tabindex="-1"></a><span class="co">#  [1] digest_0.6.33     R6_2.6.1          codetools_0.2-19  fastmap_1.2.0    </span></span>
<span id="cb15-27"><a href="#cb15-27" tabindex="-1"></a><span class="co">#  [5] xfun_0.50.6       cachem_1.1.0      knitr_1.49        pkgconfig_2.0.3  </span></span>
<span id="cb15-28"><a href="#cb15-28" tabindex="-1"></a><span class="co">#  [9] htmltools_0.5.8.1 rmarkdown_2.29    lifecycle_1.0.4   cli_3.6.2        </span></span>
<span id="cb15-29"><a href="#cb15-29" tabindex="-1"></a><span class="co"># [13] vctrs_0.6.5       sass_0.4.9        jquerylib_0.1.4   compiler_4.3.2   </span></span>
<span id="cb15-30"><a href="#cb15-30" tabindex="-1"></a><span class="co"># [17] prettyunits_1.2.0 rstudioapi_0.17.1 tools_4.3.2       hms_1.1.3        </span></span>
<span id="cb15-31"><a href="#cb15-31" tabindex="-1"></a><span class="co"># [21] evaluate_1.0.3    bslib_0.9.0       yaml_2.3.8        crayon_1.5.3     </span></span>
<span id="cb15-32"><a href="#cb15-32" tabindex="-1"></a><span class="co"># [25] rlang_1.1.2       jsonlite_1.8.9    progress_1.2.3</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
