<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Gao Wang and Shengtong Han" />

<meta name="date" content="2023-10-30" />

<title>Illustrative examples</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Illustrative examples</h1>
<h4 class="author">Gao Wang and Shengtong Han</h4>
<h4 class="date">2023-10-30</h4>



<div id="references" class="section level2">
<h2>References</h2>
<p>A Bayesian method for rare variant analysis using functional
annotations and its application to Autism: <a href="https://www.biorxiv.org/content/10.1101/828061v1" class="uri">https://www.biorxiv.org/content/10.1101/828061v1</a></p>
</div>
<div id="examples" class="section level2">
<h2>Examples</h2>
<div id="run-mirage-mirage_vs-on-toy-examples" class="section level3">
<h3>Run mirage, mirage_vs on toy examples</h3>
<p>Here is a minimal example for <code>mirage</code>. First load two
test toy data-sets, <code>mirage_vs_toy</code> is for variant set (VS)
with variant level only and <code>mirage_toy</code> is for gene set with
both variant level and gene level.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(mirage)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">data</span>(mirage_toy)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">head</span>(mirage_toy)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co">#                                 ID   Gene No.case No.contr group.index</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co"># 15129    6:161006077-161006077_C_T    LPA     298      272           1</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co"># 169        1:11205024-11205024_C_T   MTOR       1        0           2</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co"># 184        1:11272449-11272449_G_A   MTOR       0        1           2</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="co"># 194   1:11290989-11290994_CTGACT_C   MTOR       1        0           2</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="co"># 239        1:11594465-11594465_C_T PTCHD2       0        1           2</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co"># 356     1:19441304-19441307_CCTT_C   UBR4       2        0           2</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="fu">data</span>(mirage_vs_toy)</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="fu">head</span>(mirage_vs_toy)</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="co">#   No.case No.contr group.index</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="co"># 1       1        0           1</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="co"># 2       3        1           1</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="co"># 3       2        4           1</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="co"># 4       1        3           1</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="co"># 5       1        0           1</span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="co"># 6       2        0           1</span></span></code></pre></div>
<p>In both <code>mirage_vs_toy</code> and <code>mirage_toy</code>, every
row is a variant with its summary statistics (<code>No.case</code>, how
many times the variant appears in cases, <code>No.contr</code>, how many
times the variant appears in controls — you can compute these quantities
from your data), variant group index <code>group.index</code> (which
variant group it is from) and gene <code>Gene</code> (if any, which gene
it is in) in columns. Variant groups can be user defined, usually
depending on its annotations. For example one can define a variant group
“rare LoF” for a loss-of-function variant ( the union of stop loss, stop
gain, frameshift indels and splice site substitutions) having minor
allele frequency &lt;1%.</p>
<p>Apply <code>mirage_vs</code> and <code>mirage</code> to toy data.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>res_vs <span class="ot">=</span> <span class="fu">mirage_vs</span>(mirage_vs_toy,<span class="at">n1=</span><span class="dv">1000</span>,<span class="at">n2=</span><span class="dv">1000</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="co"># Computing LRT statistics and p-value for every variant and all as a whole ...</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co"># Computing LRT statistics and p-values by categories ...</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">str</span>(res_vs)</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co"># List of 4</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#  $ eta.est   : Named num 0.106</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#   ..- attr(*, &quot;names&quot;)= chr &quot;1&quot;</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#  $ full.info :&#39;data.frame&#39;:   213 obs. of  5 variables:</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#   ..$ ID      : int [1:213] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#   ..$ No.case : int [1:213] 1 3 2 1 1 2 5 1 1 0 ...</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co">#   ..$ No.contr: int [1:213] 0 1 4 3 0 0 4 0 2 1 ...</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co">#   ..$ category: int [1:213] 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#   ..$ var.BF  : num [1:213] 1.452 1.577 0.239 0.278 1.452 ...</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#  $ eta.pvalue: num 0.0577</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="co">#  $ post.prob : num [1:213] 0.1474 0.1581 0.0276 0.032 0.1474 ...</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>res_vs<span class="sc">$</span>eta.est  <span class="co"># parameter estimate </span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="co">#         1 </span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="co"># 0.1063957</span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a>res_vs<span class="sc">$</span>eta.pvalue <span class="co"># p value </span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="co"># [1] 0.05768078</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="fu">head</span>(res_vs<span class="sc">$</span>full.info) <span class="co"># variant information </span></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="co">#   ID No.case No.contr category    var.BF</span></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="co"># 1  1       1        0        1 1.4520839</span></span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a><span class="co"># 2  2       3        1        1 1.5772349</span></span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a><span class="co"># 3  3       2        4        1 0.2388117</span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a><span class="co"># 4  4       1        3        1 0.2775119</span></span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a><span class="co"># 5  5       1        0        1 1.4520839</span></span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a><span class="co"># 6  6       2        0        1 2.1375094</span></span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="fu">head</span>(res_vs<span class="sc">$</span>post.prob) <span class="co"># posterior probability (PP) of variant </span></span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a><span class="co"># [1] 0.14740532 0.15810118 0.02764764 0.03198473 0.14740532 0.20286933</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>res<span class="ot">=</span><span class="fu">mirage</span>(mirage_toy,<span class="at">n1=</span><span class="dv">4315</span>,<span class="at">n2=</span><span class="dv">4315</span>)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co"># Computing gene level LRT statistics and p-values ...</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co"># Computing LRT statistics and p-values by categories ...</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>res<span class="sc">$</span>delta.est</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co"># [1] 0.925347</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>res<span class="sc">$</span>delta.pvalue</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co"># [1] 0.9347734</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>res<span class="sc">$</span>eta.est</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#             1             2             3             4             5 </span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#  0.000000e+00  1.030961e-01 4.940656e-324 1.976263e-323  7.871498e-02</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>res<span class="sc">$</span>eta.pvalue</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co"># [1] 1.0000000 0.4206792 1.0000000 1.0000000 0.2782136</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="fu">head</span>(res<span class="sc">$</span>BF.PP.gene)</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#     Gene        BF post.prob</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="co"># 1    LPA 0.9164196 0.9190893</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="co"># 2   MTOR 1.0405812 0.9280489</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="co"># 3 PTCHD2 0.8327124 0.9116743</span></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="co"># 4   UBR4 1.2899155 0.9411381</span></span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a><span class="co"># 5  CSMD2 0.8010560 0.9085032</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a><span class="co"># 6   NCDN 1.0006285 0.9253904</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a><span class="fu">head</span>(res<span class="sc">$</span>BF.all[<span class="dv">1</span>])  <span class="co"># variant information in the first gene </span></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a><span class="co"># [[1]]</span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="co">#                                ID Gene No.case No.contr category       var.BF</span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a><span class="co"># 15129   6:161006077-161006077_C_T  LPA     298      272        1 1.634504e-07</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a><span class="co"># 15078 6:160966536-160966536_A_ACC  LPA       2        2        2 6.059431e-01</span></span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a><span class="co"># 15079 6:160966537-160966539_TTG_T  LPA       2        2        2 6.059431e-01</span></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a><span class="co"># 15082  6:160968821-160968822_AT_A  LPA       1        0        2 1.452084e+00</span></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a><span class="co"># 15102   6:160969584-160969584_G_C  LPA       1        4        2 1.896752e-01</span></span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a><span class="co"># 15107  6:160977184-160977185_AC_A  LPA       0        1        2 5.479161e-01</span></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a><span class="co"># 15114   6:160978560-160978560_G_A  LPA       1        0        2 1.452084e+00</span></span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a><span class="co"># 15116   6:160998167-160998167_C_T  LPA       1        5        2 1.405388e-01</span></span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a><span class="co"># 15142   6:161007480-161007480_C_T  LPA       1        0        2 1.452084e+00</span></span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a><span class="co"># 15153   6:161010638-161010638_C_T  LPA       1        0        2 1.452084e+00</span></span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a><span class="co"># 15163   6:161012014-161012014_G_T  LPA       0        1        2 5.479161e-01</span></span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a><span class="co"># 15171   6:161015102-161015102_C_A  LPA       1        0        2 1.452084e+00</span></span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a><span class="co"># 15177  6:161016494-161016494_C_CA  LPA       1        0        2 1.452084e+00</span></span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a><span class="co"># 15180   6:161020531-161020531_C_T  LPA       0        1        2 5.479161e-01</span></span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a><span class="co"># 15210   6:161071369-161071369_C_T  LPA       1        0        2 1.452084e+00</span></span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a><span class="co"># 15061   6:160952816-160952816_T_C  LPA     118      103        3 7.596685e-02</span></span>
<span id="cb3-40"><a href="#cb3-40" tabindex="-1"></a><span class="co"># 15068   6:160961137-160961137_T_C  LPA     257      248        3 3.931896e-08</span></span>
<span id="cb3-41"><a href="#cb3-41" tabindex="-1"></a><span class="co"># 15073   6:160963771-160963771_C_A  LPA      50       44        4 9.300675e-02</span></span>
<span id="cb3-42"><a href="#cb3-42" tabindex="-1"></a><span class="co"># 15074   6:160963774-160963774_C_G  LPA      21       32        4 5.040937e-02</span></span>
<span id="cb3-43"><a href="#cb3-43" tabindex="-1"></a><span class="co"># 15075   6:160963774-160963774_C_T  LPA       0        2        4 3.291740e-01</span></span>
<span id="cb3-44"><a href="#cb3-44" tabindex="-1"></a><span class="co"># 15081   6:160966559-160966559_G_A  LPA      17       28        4 5.484098e-02</span></span>
<span id="cb3-45"><a href="#cb3-45" tabindex="-1"></a><span class="co"># 15135   6:161006105-161006105_C_T  LPA      88      100        4 2.698210e-02</span></span>
<span id="cb3-46"><a href="#cb3-46" tabindex="-1"></a><span class="co"># 15148   6:161007619-161007619_G_T  LPA      26       20        4 2.857868e-01</span></span>
<span id="cb3-47"><a href="#cb3-47" tabindex="-1"></a><span class="co"># 15195   6:161022107-161022107_C_T  LPA      79       90        4 2.795875e-02</span></span>
<span id="cb3-48"><a href="#cb3-48" tabindex="-1"></a><span class="co">#       original.index</span></span>
<span id="cb3-49"><a href="#cb3-49" tabindex="-1"></a><span class="co"># 15129              1</span></span>
<span id="cb3-50"><a href="#cb3-50" tabindex="-1"></a><span class="co"># 15078              2</span></span>
<span id="cb3-51"><a href="#cb3-51" tabindex="-1"></a><span class="co"># 15079              2</span></span>
<span id="cb3-52"><a href="#cb3-52" tabindex="-1"></a><span class="co"># 15082              2</span></span>
<span id="cb3-53"><a href="#cb3-53" tabindex="-1"></a><span class="co"># 15102              2</span></span>
<span id="cb3-54"><a href="#cb3-54" tabindex="-1"></a><span class="co"># 15107              2</span></span>
<span id="cb3-55"><a href="#cb3-55" tabindex="-1"></a><span class="co"># 15114              2</span></span>
<span id="cb3-56"><a href="#cb3-56" tabindex="-1"></a><span class="co"># 15116              2</span></span>
<span id="cb3-57"><a href="#cb3-57" tabindex="-1"></a><span class="co"># 15142              2</span></span>
<span id="cb3-58"><a href="#cb3-58" tabindex="-1"></a><span class="co"># 15153              2</span></span>
<span id="cb3-59"><a href="#cb3-59" tabindex="-1"></a><span class="co"># 15163              2</span></span>
<span id="cb3-60"><a href="#cb3-60" tabindex="-1"></a><span class="co"># 15171              2</span></span>
<span id="cb3-61"><a href="#cb3-61" tabindex="-1"></a><span class="co"># 15177              2</span></span>
<span id="cb3-62"><a href="#cb3-62" tabindex="-1"></a><span class="co"># 15180              2</span></span>
<span id="cb3-63"><a href="#cb3-63" tabindex="-1"></a><span class="co"># 15210              2</span></span>
<span id="cb3-64"><a href="#cb3-64" tabindex="-1"></a><span class="co"># 15061              3</span></span>
<span id="cb3-65"><a href="#cb3-65" tabindex="-1"></a><span class="co"># 15068              3</span></span>
<span id="cb3-66"><a href="#cb3-66" tabindex="-1"></a><span class="co"># 15073              4</span></span>
<span id="cb3-67"><a href="#cb3-67" tabindex="-1"></a><span class="co"># 15074              4</span></span>
<span id="cb3-68"><a href="#cb3-68" tabindex="-1"></a><span class="co"># 15075              4</span></span>
<span id="cb3-69"><a href="#cb3-69" tabindex="-1"></a><span class="co"># 15081              4</span></span>
<span id="cb3-70"><a href="#cb3-70" tabindex="-1"></a><span class="co"># 15135              4</span></span>
<span id="cb3-71"><a href="#cb3-71" tabindex="-1"></a><span class="co"># 15148              4</span></span>
<span id="cb3-72"><a href="#cb3-72" tabindex="-1"></a><span class="co"># 15195              4</span></span></code></pre></div>
<p><code>n1, n2</code> are sample sizes in cases and controls. Other
parameters are set by default. Note that the input data
<code>mirage_toy, mirage_vs_toy</code> must be in data.frame format
without missing values. <code>BF.all</code> has all variant level
information for a gene.</p>
</div>
<div id="calculate-bayes-factor-of-every-variant-group-in-a-gene" class="section level3">
<h3>Calculate Bayes factor of every variant group in a gene</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="do">########## after run MIRAGE,  given a gene, calculate the BF of every variant group.  </span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>res<span class="ot">=</span><span class="fu">mirage</span>(mirage_toy,<span class="at">n1=</span><span class="dv">4315</span>,<span class="at">n2=</span><span class="dv">4315</span>)  <span class="co"># run MIRAGE on a sample </span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co"># Computing gene level LRT statistics and p-values ...</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co"># Computing LRT statistics and p-values by categories ...</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>num_category<span class="ot">=</span><span class="dv">6</span> <span class="co"># number of variant groups </span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>variant_full_infor<span class="ot">=</span>res<span class="sc">$</span>BF.all[[<span class="dv">1</span>]]  <span class="co"># given a gene, say the first gene </span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>variant_group_BF<span class="ot">=</span><span class="fu">numeric</span>() <span class="co"># store variant group BF for a given gene </span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>num_category)</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>{</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(variant_full_infor<span class="sc">$</span>original.index<span class="sc">==</span>j)<span class="sc">==</span><span class="dv">0</span>) <span class="co"># no variants in this group</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>    variant_group_BF[j]<span class="ot">=</span><span class="cn">NA</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>  </span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(variant_full_infor<span class="sc">$</span>original.index<span class="sc">==</span>j)<span class="sc">&gt;</span><span class="dv">0</span>) <span class="co"># variants exist in this group</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>    variant_group_BF[j]<span class="ot">=</span><span class="fu">prod</span>(<span class="dv">1</span><span class="sc">-</span>res<span class="sc">$</span>eta.est[j]<span class="sc">+</span>res<span class="sc">$</span>eta.est[j]<span class="sc">*</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>    variant_full_infor[<span class="fu">which</span>(variant_full_infor<span class="sc">$</span>original.index<span class="sc">==</span>j),]<span class="sc">$</span>var.BF) </span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a>  <span class="co"># product of BF of variants within that variant group </span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a>   </span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>}</span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a>variant_group_BF</span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="co"># [1] 1.0000000 0.9164266 1.0000000 1.0000000        NA        NA</span></span></code></pre></div>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<p>Here are some details about the computing environment, including the
versions of R, and the R packages, used to generate these results.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co"># R version 4.2.2 (2022-10-31 ucrt)</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co"># Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co"># Running under: Windows 10 x64 (build 19045)</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co"># Matrix products: default</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co"># locale:</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co"># [1] LC_COLLATE=English_United States.utf8 </span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co"># [2] LC_CTYPE=English_United States.utf8   </span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co"># [3] LC_MONETARY=English_United States.utf8</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="co"># [4] LC_NUMERIC=C                          </span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="co"># [5] LC_TIME=English_United States.utf8    </span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a><span class="co"># attached base packages:</span></span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a><span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a><span class="co"># other attached packages:</span></span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a><span class="co"># [1] mirage_0.1.0.0</span></span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb5-21"><a href="#cb5-21" tabindex="-1"></a><span class="co"># loaded via a namespace (and not attached):</span></span>
<span id="cb5-22"><a href="#cb5-22" tabindex="-1"></a><span class="co">#  [1] rstudioapi_0.14   knitr_1.42        hms_1.1.2         progress_1.2.2   </span></span>
<span id="cb5-23"><a href="#cb5-23" tabindex="-1"></a><span class="co">#  [5] R6_2.5.1          rlang_1.0.6       fastmap_1.1.0     tools_4.2.2      </span></span>
<span id="cb5-24"><a href="#cb5-24" tabindex="-1"></a><span class="co">#  [9] xfun_0.37         cli_3.4.1         jquerylib_0.1.4   htmltools_0.5.4  </span></span>
<span id="cb5-25"><a href="#cb5-25" tabindex="-1"></a><span class="co"># [13] ellipsis_0.3.2    yaml_2.3.7        digest_0.6.31     lifecycle_1.0.3  </span></span>
<span id="cb5-26"><a href="#cb5-26" tabindex="-1"></a><span class="co"># [17] crayon_1.5.2      sass_0.4.5        vctrs_0.5.2       codetools_0.2-18 </span></span>
<span id="cb5-27"><a href="#cb5-27" tabindex="-1"></a><span class="co"># [21] cachem_1.0.7      evaluate_0.20     rmarkdown_2.20    compiler_4.2.2   </span></span>
<span id="cb5-28"><a href="#cb5-28" tabindex="-1"></a><span class="co"># [25] bslib_0.4.2       prettyunits_1.1.1 jsonlite_1.8.4    pkgconfig_2.0.3</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
