---
title: "Illustrative examples"
author: "Gao Wang and Shengtong Han"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{minimal example}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(collapse = TRUE,comment = "#",fig.width = 4.5,
                      fig.height = 3,fig.align = "center",
                      fig.cap = "&nbsp;",dpi = 120)
```



## References

A Bayesian method for rare variant analysis using functional annotations and its application to Autism: https://www.biorxiv.org/content/10.1101/828061v1

## Examples


### Run mirage, mirage_vs on toy examples 

Here is a minimal example for `mirage`. First load  two test toy data-sets, `mirage_vs_toy` is for variant set (VS) with variant level only and `mirage_toy` is for gene set with both variant level and gene level.

```{r}
library(mirage)
data(mirage_toy)
head(mirage_toy)
data(mirage_vs_toy)
head(mirage_vs_toy)
```
In both `mirage_vs_toy` and `mirage_toy`, every row is a variant with its summary statistics (`No.case`, how many times the variant appears in cases, `No.contr`, how many times the variant appears in controls --- you can compute these quantities from your data), variant group index `group.index` (which variant group it is from) and gene `Gene` (if any, which gene it is in) in columns. Variant groups can be user defined, usually depending on its annotations. For example one can define a variant group "rare LoF" for a loss-of-function variant ( the union of stop loss, stop gain, frameshift indels and splice site substitutions) having minor allele frequency <1%.

Apply `mirage_vs` and `mirage` to toy data.

```{r}
res_vs = mirage_vs(mirage_vs_toy,n1=1000,n2=1000)
str(res_vs)
res_vs$eta.est  # parameter estimate 
res_vs$eta.pvalue # p value 
head(res_vs$full.info) # variant information 
head(res_vs$post.prob) # posterior probability (PP) of variant 
```


```{r, cache=T}
res=mirage(mirage_toy,n1=4315,n2=4315)
res$delta.est
res$delta.pvalue
res$eta.est
res$eta.pvalue
head(res$BF.PP.gene)
head(res$BF.all[1])  # variant information in the first gene 
```



`n1, n2` are sample sizes in cases and controls. Other parameters are set by default. Note that the input data `mirage_toy, mirage_vs_toy` must be in data.frame  format without missing values. `BF.all` has all variant level information for a gene. 


### Calculate Bayes factor of every variant group in a gene 

```{r, cache=T}
########## after run MIRAGE,  given a gene, calculate the BF of every variant group.  
res=mirage(mirage_toy,n1=4315,n2=4315)  # run MIRAGE on a sample 
num_category=6 # number of variant groups 
variant_full_infor=res$BF.all[[1]]  # given a gene, say the first gene 
variant_group_BF=numeric() # store variant group BF for a given gene 
for (j in 1:num_category)
{
  if (sum(variant_full_infor$original.index==j)==0) # no variants in this group
    variant_group_BF[j]=NA
  
  if (sum(variant_full_infor$original.index==j)>0) # variants exist in this group
    variant_group_BF[j]=prod(1-res$eta.est[j]+res$eta.est[j]*
    variant_full_infor[which(variant_full_infor$original.index==j),]$var.BF) 
  # product of BF of variants within that variant group 
   
}
variant_group_BF
```























## Session information

Here are some details about the computing environment, including the
versions of R, and the R packages, used to generate these results.

```{r}
sessionInfo()
```
