<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Illustrative examples • mirage</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Illustrative examples">
<meta property="og:description" content="mirage">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mirage</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/mwe.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../articles/methodtutorial.html">Method Tutorial</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/xinhe-lab/mirage" class="external-link">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Illustrative examples</h1>
                        <h4 data-toc-skip class="author">Gao Wang,
Shengtong Han, Xiaotong Sun</h4>
            
            <h4 data-toc-skip class="date">2024-04-14</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/xinhe-lab/mirage/blob/HEAD/vignettes/mwe.Rmd" class="external-link"><code>vignettes/mwe.Rmd</code></a></small>
      <div class="hidden name"><code>mwe.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">'xinhe-lab/mirage'</span><span class="op">)</span></span></code></pre></div>
<p>Load the package</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/xinhe-lab/mirage" class="external-link">mirage</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preparing-input-data">Preparing input data<a class="anchor" aria-label="anchor" href="#preparing-input-data"></a>
</h2>
<div class="section level3">
<h3 id="summary-statistics-for-each-rare-variant-">Summary statistics for each rare variant.<a class="anchor" aria-label="anchor" href="#summary-statistics-for-each-rare-variant-"></a>
</h3>
<p>MIRAGE requires summary statistics as input, including the ID for
each rare variant. For each variant, it needs the allele counts in both
cases and controls, the gene to which it belongs, and the annotation
group index it is associated with. The annotation groups should be
distinct categories, such as loss-of-function (LoF) variants and
missense variants predicted to have deleterious effects.</p>
<p>We include an example input for MIRAGE in our package, which users
can reference to format their own data accordingly.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">mirage_toy</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mirage_toy</span><span class="op">)</span></span>
<span><span class="co">#                                 ID   Gene No.case No.contr group.index</span></span>
<span><span class="co"># 15129    6:161006077-161006077_C_T    LPA     298      272           1</span></span>
<span><span class="co"># 169        1:11205024-11205024_C_T   MTOR       1        0           2</span></span>
<span><span class="co"># 184        1:11272449-11272449_G_A   MTOR       0        1           2</span></span>
<span><span class="co"># 194   1:11290989-11290994_CTGACT_C   MTOR       1        0           2</span></span>
<span><span class="co"># 239        1:11594465-11594465_C_T PTCHD2       0        1           2</span></span>
<span><span class="co"># 356     1:19441304-19441307_CCTT_C   UBR4       2        0           2</span></span></code></pre></div>
<p>Each row in the input represents a variant with its summary
statistics:</p>
<ul>
<li>
<code>No.case</code>: Number of times the variant appears in
cases.</li>
<li>
<code>No.contr</code>: Number of times the variant appears in
controls.</li>
<li>
<code>group.index</code>: Indicates the variant group it belongs
to.</li>
<li>
<code>Gene</code>: Specifies the associated gene, if any.</li>
</ul>
<p>Variant groups are user-defined, usually based on annotations. For
example, a group “rare LoF” could be defined for loss-of-function
variants (including stop loss, stop gain, frameshift indels, and splice
site substitutions) with a minor allele frequency of less than 1%.</p>
</div>
</div>
<div class="section level2">
<h2 id="additional-arguments">Additional arguments<a class="anchor" aria-label="anchor" href="#additional-arguments"></a>
</h2>
<p>MIRAGE requires the following additional arguments:</p>
<ul>
<li>
<code>n1</code>: Sample size in cases (required).</li>
<li>
<code>n2</code>: Sample size in controls (required).</li>
<li>
<code>gamma</code>: A list of category-specific hyper prior shape
parameters in the Beta distribution for effect size, or a numeric value
if all categories share the same effect size. Default value is 3.</li>
<li>
<code>sigma</code>: A list of category-specific hyper prior scale
parameters in the Beta distribution for effect size, or a numeric value
if all categories share the same effect size. Default value is 2.</li>
<li>
<code>eta.init</code>: Initial value for the prior on the proportion
of risk variants in a variant set. Default value is 0.1.</li>
<li>
<code>delta.init</code>: Initial value for the prior on the
proportion of risk genes. Must be a positive number between 0 and 1.
Default value is 0.1.</li>
<li>
<code>estimate.delta</code>: Specifies whether <code>delta</code>
should be estimated (<code>TRUE</code>) or fixed at
<code>delta.init</code> (<code>FALSE</code>). Default is
<code>TRUE</code>.</li>
<li>
<code>max.iter</code>: Maximum number of iterations, enforcing the
EM algorithm to stop. Default is 10,000.</li>
<li>
<code>tol</code>: Threshold of parameter estimate difference to
determine the convergence of the EM algorithm. Default value is
1e-05.</li>
<li>
<code>verbose</code>: When <code>TRUE</code>, enables detailed
output during computation. Default is <code>TRUE</code>.</li>
</ul>
<p>Only <code>n1</code> and <code>n2</code> are required to run MIRAGE;
other parameters have default values.</p>
</div>
<div class="section level2">
<h2 id="running-mirage-on-toy-data">Running MIRAGE on toy data<a class="anchor" aria-label="anchor" href="#running-mirage-on-toy-data"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span><span class="op">=</span><span class="fu"><a href="../reference/mirage.html">mirage</a></span><span class="op">(</span><span class="va">mirage_toy</span>,n1<span class="op">=</span><span class="fl">4315</span>,n2<span class="op">=</span><span class="fl">4315</span><span class="op">)</span></span>
<span><span class="co"># Computing gene level LRT statistics and p-values ...</span></span>
<span><span class="co"># Computing LRT statistics and p-values by categories ...</span></span></code></pre></div>
<p>The output of MIRAGE is a list containing the following values:</p>
<ul>
<li>
<code>BF.PP.gene</code>: Bayes factor and posterior probability of
genes.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">BF.PP.gene</span><span class="op">)</span></span>
<span><span class="co">#     Gene        BF post.prob</span></span>
<span><span class="co"># 1    LPA 0.9164196 0.9190893</span></span>
<span><span class="co"># 2   MTOR 1.0405812 0.9280489</span></span>
<span><span class="co"># 3 PTCHD2 0.8327124 0.9116743</span></span>
<span><span class="co"># 4   UBR4 1.2899155 0.9411381</span></span>
<span><span class="co"># 5  CSMD2 0.8010560 0.9085032</span></span>
<span><span class="co"># 6   NCDN 1.0006285 0.9253904</span></span></code></pre></div>
<ul>
<li>
<code>BF.all</code>: A list of Bayes factors for all variants within
a gene.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">BF.all</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="co"># variant information in the first gene </span></span>
<span><span class="co"># [[1]]</span></span>
<span><span class="co">#                                ID Gene No.case No.contr category       var.BF</span></span>
<span><span class="co"># 15129   6:161006077-161006077_C_T  LPA     298      272        1 1.634504e-07</span></span>
<span><span class="co"># 15078 6:160966536-160966536_A_ACC  LPA       2        2        2 6.059431e-01</span></span>
<span><span class="co"># 15079 6:160966537-160966539_TTG_T  LPA       2        2        2 6.059431e-01</span></span>
<span><span class="co"># 15082  6:160968821-160968822_AT_A  LPA       1        0        2 1.452084e+00</span></span>
<span><span class="co"># 15102   6:160969584-160969584_G_C  LPA       1        4        2 1.896752e-01</span></span>
<span><span class="co"># 15107  6:160977184-160977185_AC_A  LPA       0        1        2 5.479161e-01</span></span>
<span><span class="co"># 15114   6:160978560-160978560_G_A  LPA       1        0        2 1.452084e+00</span></span>
<span><span class="co"># 15116   6:160998167-160998167_C_T  LPA       1        5        2 1.405388e-01</span></span>
<span><span class="co"># 15142   6:161007480-161007480_C_T  LPA       1        0        2 1.452084e+00</span></span>
<span><span class="co"># 15153   6:161010638-161010638_C_T  LPA       1        0        2 1.452084e+00</span></span>
<span><span class="co"># 15163   6:161012014-161012014_G_T  LPA       0        1        2 5.479161e-01</span></span>
<span><span class="co"># 15171   6:161015102-161015102_C_A  LPA       1        0        2 1.452084e+00</span></span>
<span><span class="co"># 15177  6:161016494-161016494_C_CA  LPA       1        0        2 1.452084e+00</span></span>
<span><span class="co"># 15180   6:161020531-161020531_C_T  LPA       0        1        2 5.479161e-01</span></span>
<span><span class="co"># 15210   6:161071369-161071369_C_T  LPA       1        0        2 1.452084e+00</span></span>
<span><span class="co"># 15061   6:160952816-160952816_T_C  LPA     118      103        3 7.596685e-02</span></span>
<span><span class="co"># 15068   6:160961137-160961137_T_C  LPA     257      248        3 3.931896e-08</span></span>
<span><span class="co"># 15073   6:160963771-160963771_C_A  LPA      50       44        4 9.300675e-02</span></span>
<span><span class="co"># 15074   6:160963774-160963774_C_G  LPA      21       32        4 5.040937e-02</span></span>
<span><span class="co"># 15075   6:160963774-160963774_C_T  LPA       0        2        4 3.291740e-01</span></span>
<span><span class="co"># 15081   6:160966559-160966559_G_A  LPA      17       28        4 5.484098e-02</span></span>
<span><span class="co"># 15135   6:161006105-161006105_C_T  LPA      88      100        4 2.698210e-02</span></span>
<span><span class="co"># 15148   6:161007619-161007619_G_T  LPA      26       20        4 2.857868e-01</span></span>
<span><span class="co"># 15195   6:161022107-161022107_C_T  LPA      79       90        4 2.795875e-02</span></span>
<span><span class="co">#       original.index</span></span>
<span><span class="co"># 15129              1</span></span>
<span><span class="co"># 15078              2</span></span>
<span><span class="co"># 15079              2</span></span>
<span><span class="co"># 15082              2</span></span>
<span><span class="co"># 15102              2</span></span>
<span><span class="co"># 15107              2</span></span>
<span><span class="co"># 15114              2</span></span>
<span><span class="co"># 15116              2</span></span>
<span><span class="co"># 15142              2</span></span>
<span><span class="co"># 15153              2</span></span>
<span><span class="co"># 15163              2</span></span>
<span><span class="co"># 15171              2</span></span>
<span><span class="co"># 15177              2</span></span>
<span><span class="co"># 15180              2</span></span>
<span><span class="co"># 15210              2</span></span>
<span><span class="co"># 15061              3</span></span>
<span><span class="co"># 15068              3</span></span>
<span><span class="co"># 15073              4</span></span>
<span><span class="co"># 15074              4</span></span>
<span><span class="co"># 15075              4</span></span>
<span><span class="co"># 15081              4</span></span>
<span><span class="co"># 15135              4</span></span>
<span><span class="co"># 15148              4</span></span>
<span><span class="co"># 15195              4</span></span></code></pre></div>
<ul>
<li>
<code>delta.est</code>: Estimate for the proportion of risk
genes.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span><span class="op">$</span><span class="va">delta.est</span></span>
<span><span class="co"># [1] 0.925347</span></span></code></pre></div>
<ul>
<li>
<code>delta.pvalue</code>: Significance test for delta = 0.</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span><span class="op">$</span><span class="va">delta.pvalue</span></span>
<span><span class="co"># [1] 0.9347734</span></span></code></pre></div>
<ul>
<li>
<code>eta.est</code>: Estimate for the proportion of risk variants
in a variant group.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span><span class="op">$</span><span class="va">eta.est</span></span>
<span><span class="co">#             1             2             3             4             5 </span></span>
<span><span class="co">#  0.000000e+00  1.030961e-01 4.940656e-324 1.976263e-323  7.871498e-02</span></span></code></pre></div>
<ul>
<li>
<code>eta.pvalue</code>: Significance test for eta = 0.</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span><span class="op">$</span><span class="va">eta.pvalue</span></span>
<span><span class="co"># [1] 1.0000000 0.4206792 1.0000000 1.0000000 0.2782136</span></span></code></pre></div>
<div class="section level3">
<h3 id="calculating-bayes-factor-for-each-variant-group-within-a-gene">Calculating Bayes factor for each variant group within a gene<a class="anchor" aria-label="anchor" href="#calculating-bayes-factor-for-each-variant-group-within-a-gene"></a>
</h3>
<p>Users may use the script below to compute Bayes factor for each
variant group within a gene. We here show an example for computing for
the first gene in our example data “LPA”.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">########## after run MIRAGE,  given a gene, calculate the BF of every variant group.  </span></span>
<span></span>
<span><span class="va">num_category</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">mirage_toy</span><span class="op">[</span><span class="va">mirage_toy</span><span class="op">$</span><span class="va">Gene</span> <span class="op">==</span><span class="st">"LPA"</span>,<span class="op">]</span><span class="op">$</span><span class="va">group.index</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="co"># number of variant groups </span></span>
<span><span class="va">variant_full_infor</span><span class="op">=</span><span class="va">res</span><span class="op">$</span><span class="va">BF.all</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>  <span class="co"># extract results for first gene </span></span>
<span><span class="va">variant_group_BF</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">numeric</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># store variant group BF for a given gene </span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">num_category</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">variant_full_infor</span><span class="op">$</span><span class="va">original.index</span><span class="op">==</span><span class="va">j</span><span class="op">)</span><span class="op">==</span><span class="fl">0</span><span class="op">)</span> <span class="co"># no variants in this group</span></span>
<span>    <span class="va">variant_group_BF</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">=</span><span class="cn">NA</span></span>
<span>  </span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">variant_full_infor</span><span class="op">$</span><span class="va">original.index</span><span class="op">==</span><span class="va">j</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span> <span class="co"># variants exist in this group</span></span>
<span>    <span class="va">variant_group_BF</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/prod.html" class="external-link">prod</a></span><span class="op">(</span><span class="fl">1</span><span class="op">-</span><span class="va">res</span><span class="op">$</span><span class="va">eta.est</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">+</span><span class="va">res</span><span class="op">$</span><span class="va">eta.est</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">*</span></span>
<span>    <span class="va">variant_full_infor</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">variant_full_infor</span><span class="op">$</span><span class="va">original.index</span><span class="op">==</span><span class="va">j</span><span class="op">)</span>,<span class="op">]</span><span class="op">$</span><span class="va">var.BF</span><span class="op">)</span> </span>
<span>  <span class="co"># product of BF of variants within that variant group </span></span>
<span>   </span>
<span><span class="op">}</span></span>
<span><span class="va">variant_group_BF</span></span>
<span><span class="co"># [1] 1.0000000 0.9164266 1.0000000 1.0000000</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>Here are some details about the computing environment, including the
versions of R, and the R packages, used to generate these results.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co"># R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co"># Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co"># Running under: CentOS Linux 7 (Core)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># Matrix products: default</span></span>
<span><span class="co"># BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># locale:</span></span>
<span><span class="co">#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co"># [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># attached base packages:</span></span>
<span><span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># other attached packages:</span></span>
<span><span class="co"># [1] mirage_0.1.0.0</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># loaded via a namespace (and not attached):</span></span>
<span><span class="co">#  [1] rstudioapi_0.13   knitr_1.39        magrittr_2.0.3    hms_1.1.1        </span></span>
<span><span class="co">#  [5] progress_1.2.2    R6_2.5.1          ragg_1.2.2        rlang_1.1.2      </span></span>
<span><span class="co">#  [9] fastmap_1.1.0     stringr_1.5.1     tools_4.2.0       xfun_0.41        </span></span>
<span><span class="co"># [13] cli_3.6.1         jquerylib_0.1.4   ellipsis_0.3.2    htmltools_0.5.2  </span></span>
<span><span class="co"># [17] systemfonts_1.0.4 yaml_2.3.5        digest_0.6.29     lifecycle_1.0.4  </span></span>
<span><span class="co"># [21] crayon_1.5.1      pkgdown_2.0.7     textshaping_0.3.6 purrr_1.0.2      </span></span>
<span><span class="co"># [25] codetools_0.2-18  sass_0.4.1        vctrs_0.6.5       fs_1.5.2         </span></span>
<span><span class="co"># [29] memoise_2.0.1     glue_1.6.2        cachem_1.0.6      evaluate_0.15    </span></span>
<span><span class="co"># [33] rmarkdown_2.25    stringi_1.7.6     compiler_4.2.0    bslib_0.3.1      </span></span>
<span><span class="co"># [37] prettyunits_1.1.1 desc_1.4.3        jsonlite_1.8.0    pkgconfig_2.0.3</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Shengtong Han, Gao Wang, Xin He.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
